clib/cMap.ml

```diff
@@ -65,6 +65,7 @@ module type ExtS = sig
 
   val fold_left : (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
   val fold_right : (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
+  val fold_left_map : (key -> 'a -> 'b -> 'a * 'c) -> 'a -> 'b t -> 'a * 'c t
```

Comment:
I'm warry that having both `fold_left` and `fold_left_map` in the same module with different conventions might be confusing.

Reply:
Both are possible. I'd expect changing `fold_left_map` to be much easier. In any case, if changing `fold_left`, this should be done in its own PR so that everyone gets the opportunity to see/comment the change.

Reply:
`CMap.fold_left` was commited by @ppedrot. Both orders make sense, though it is difficult to evaluate whether adopting the convention from `CList` should lead to
```ocaml
fold_left_map : (key -> 'a -> 'b -> 'a * 'c) -> 'a -> 'b t -> 'a * 'c t
```
or to
```ocaml
fold_left_map : ( 'a -> key -> 'b -> 'a * 'c) -> 'a -> 'b t -> 'a * 'c t
```
And in case we adopt the convention in `CMap`, it is not fully clear either whether it should be
```ocaml
fold_left_map : (key -> 'a -> 'b -> 'b * 'c) -> 'a -> 'b t -> 'b * 'c t
```
or
```ocaml
fold_left_map : (key -> 'a -> 'b -> 'c * 'b) -> 'a -> 'b t -> 'c t * 'b
```
I opened a separate PR which uses the 4th choice (and adds `fold_right_map`).

Reply:
Thanks, that should make the discussion and decision easier on this point.

Reply:
Done as part of https://github.com/coq/coq/pull/19561